### PR TITLE
*: build and use count-min sketch

### PR DESCRIPTION
**Pull request by lamxTyler**
This PR builds count-min sketch when analyze and use it for point queries. Especially, we do not build it for primary key when it is handle and the unique index.
PTAL @coocood  @hanfei1991 @winoros @zz-jason

### CODE REVIEW

File: statistics/cmsketch.go [hunk 1]
```diff
@@ -55,11 +54,15 @@ func (c *CMSketch) insert(val *types.Datum) error {
 	return nil
 }
 
-func (c *CMSketch) query(val *types.Datum) (uint32, error) {
+func (c *CMSketch) queryValue(val *types.Datum) (uint32, error) {
```
Comment (coocood) [inline statistics/cmsketch.go]: We don't need to use pointer for `val`.

**Comment by shenli** [comment]
Should we merge this PR after TiKV support count-min sketch?

**Comment by lamxTyler** [comment]
@shenli We can, but do not have to, because the code does not assume that tikv will build count-min sketch despite we send the request.

File: statistics/cmsketch.go (hunk 1 above)
Comment (zz-jason) [reply statistics/cmsketch.go]: I think it's better to use pointer to avoid copying big object.

**Comment by coocood** [comment]
LGTM

**Pull request merged**